lib/Target/LLVMIR/LLVMIRTranslation.cpp

```diff
@@ -83,7 +83,7 @@ static void amendLLVMFunc(llvm::Function *func, const NVVMMetadata &metadata,
   if (metadata.isKernel) {
     if (isROCM) {
       func->setCallingConv(llvm::CallingConv::AMDGPU_KERNEL);
-      func->addFnAttr("amdgpu-flat-work-group-size", "1, 1024");
+      func->addFnAttr("amdgpu-flat-work-group-size", "1, " + std::to_string(numWarps*64));
```

Comment:
We have hardware that have waveSize of 32, and wave size can be got from `mod`. There are many examples in this PR https://github.com/ROCmSoftwarePlatform/triton/pull/244.